app/Models/FreePages.php

```diff
@@ -51,23 +53,73 @@ class FreePages extends Model
         'comment',
     ];
 
-    public function user()
+    protected $casts = [
+        'deadline' => 'date',
+    ];
+
+    /**
+     * The user this free pages entry belongs to.
+     * @return BelongsTo
+     */
+    public function user(): BelongsTo
     {
-        return $this->belongsTo('App\Models\User');
+        return $this->belongsTo(User::class);
     }
 
+    /**
+     * The print account this free pages entry belongs to.
+     * @return BelongsTo
+     */
     public function printAccount()
     {
-        return $this->belongsTo('App\Models\PrintAccount', 'user_id', 'user_id');
+        return $this->belongsTo(PrintAccount::class, 'user_id', 'user_id');
+    }
+
+    /**
+     * Wether the free pages are still available.
+     * @return bool
+     */
+    protected function available()
```

Comment:
This fucntion is called outside from `printing.blade.php`, should not be protected, it should be public.

Reply:
AFAIK attributes don't care about the visibility of the function behind it. Virtually every attribute in the project is created in a protected function.

Reply:
For me route `/profile` was crashing with error "App\Models\FreePages::available must return a relationship instance.". The return value type should be included according to the documentation (https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting): "All attribute accessor / mutator methods must declare a return type-hint of `Illuminate\Database\Eloquent\Casts\Attribute`":
```php
protected function available() : Attribute
```

Reply:
With this change, it works fine on my machine

Reply:
Ah so it needs the type hint, okay. That's why I don't like use the new syntax for attribute creation @kdmnk :P. But thanks, will fix it tomorrow

Reply:
Thanks, there are also other instances where it should be fixed (like `App\Models\PrintJob::translatedCost`).